tests/vmi_multus_test.go

```diff
@@ -934,13 +936,13 @@ func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress s
 	Expect(err).ToNot(HaveOccurred(), "MAC %q was not found in the VMI %s within the given timeout", macAddress, vmi.Name)
 }
 
-func pingVirtualMachine(vmi *v1.VirtualMachineInstance, ipAddr, prompt string) {
+func testPingFromVMI(vmi *v1.VirtualMachineInstance, ipAddr, prompt string) error {
```

Comment:
> The thing is I would like the error to include also whatever returns from the cmd-expecter,
> I can maybe try to modify the error in testPingFromVMI(...). something like:
> 
>   `return fmt.Errorf("VMI %s failed to ping IP %s with an error %v", detachedVMI.Name, testIP, err)`
> 
I'm ok with that, although you do not need "with an error ..." but a simple "... , error: %v".

> though I am not sure why this is necessary. formatting within the Expect is done all over.

You created a helper, which does a ping and has all the needed info, I see no reason to repeat this all over.

> if an error occurs it prints the whole error object , including the message.
> maybe it is just me, but I find it a bit more annoying to read this high level information from there.

Why is it high level? The helper does exactly what you error about, it is at the correct abstraction layer.
The proof for that is that it had all the data already and you do not add anything more above.

> furthermore , this high level message won't show up in the summery, at the end of the run, which can be also less convenient.

I'm not sure I follow here.
You are saying that the message is included in some summary when the test ends? While the err itself does not?

I mainly do not like to add text in the test because it annoys the eye. I want a clean reading, and hoping that the test framework will include the relevant info when failing (that occurs when the error is printed).

Reply:
Regarding adding text to `Expect` errors, although it might annoy the eye sometimes it is very important to add those messages.
Many times I found myself getting not informative error messages in ginkgo tests, it confusing and make you waste time on debugging and on reproducing that error, for example:
I got `AfterSuite` error where a namespace failed to delete, I had to add this message manually to this function 
https://github.com/kubevirt/kubevirt/blob/master/tests/utils.go#L1692
just to get the reason why that namespace failed to delete.
There are more examples like these.
Also, the tests code has minimal documentation (with comment) and we barely use `By`, so we should at least add `Expect` messages to understand better what we actually expect to happen during the tests.

Reply:
I am not arguing against information, I argue against duplicate info.

If the info can be provided through the error it should be preferred so we do not need to repeat it per each assert.
If “what” is asserted can be expressed in the variables/struct names it should be preferred.
If there is a need for more context that is not in the error or the variable names, add the text next to the assert.
You should trust code before you trust text, make an effort to write readable code over overloading it with text, especially when the text does not add any new info.